share/wake/lib/core/json.wake

```diff
@@ -538,3 +538,106 @@ export def lastValueInList (values: List JValue): Result JValue Error =
             | flatten
             | deduplicateJObjectKeys lastValueInList
             | rmap JObject
+
+export def jString: JValue => List String = match _
+    JString s = |> s
+    _         = Nil
+
+export def editJString (f: String => String): JValue => JValue = match _
+    JString s = JString (f s)
+    x         = x
+
+export def jInteger: JValue => List Integer = match _
+    JInteger i = |> i
+    _          = Nil
+
+export def editJInteger (f: Integer => Integer): JValue => JValue = match _
+    JInteger i = JInteger (f i)
+    x          = x
+
+export def jBoolean: JValue => List Boolean = match _
+    JBoolean b = |> b
+    _          = Nil
+
+export def editJBoolean (f: Boolean => Boolean): JValue => JValue = match _
+    JBoolean b = JBoolean (f b)
+    x          = x
+
+export def jDouble: JValue => List Double = match _
+    JDouble d = |> d
+    _         = Nil
+
+export def editJDouble (f: Double => Double): JValue => JValue = match _
+    JDouble d = JDouble (f d)
+    x         = x
+
+export def jNumber: JValue => List Double = match _
+    JDouble x  = |> x
+    JInteger x = |> x | mapPartial dint
+    _          = Nil
+
+export def editJNumber (f: Double => Double): JValue => JValue = match _
+    JDouble x  = JDouble (f x)
+    JInteger x = match (dint x)
+        Some dx = JDouble (f dx)
+        None    = JInteger x
+    x          = x
+
+export def jNull: JValue => List Unit = match _
+    JNull = |> Unit
+    _     = Nil
+
+export def editJNull (f: Unit => Unit): JValue => JValue = match _
+    JNull =
+        def _ = f Unit
+        JNull
+    x     = x
+
+export def jObject: JValue => List (List (Pair String JValue)) = match _
+    JObject obj = |> obj
+    _           = Nil
+
+export def editJObject (f: List (Pair String JValue) => List (Pair String JValue)): JValue => JValue = match _
+    JObject obj = JObject (f obj)
+    x           = x
+
+export def jArray: JValue => List (List JValue) = match _
+    JArray arr = |> arr
+    _          = Nil
+
+export def editJArray (f: List JValue => List JValue): JValue => JValue = match _
+    JArray arr = JArray (f arr)
+    x          = x
+
+export def jFlatten: JValue => List JValue = _ | jArray |> (_)
+
+export def editJFlatten: (JValue => JValue) => JValue => JValue = editJArray $ map $ _
+
+export def jAt (i: Integer): JValue => List JValue =
+    jFlatten _ | at i | optionToList
+
+def atE (i: Integer): (a => a) => List a => List a =
```

Comment:
Or just call it `editAt` and move it to `list.wake`. I'm fine either way.

Reply:
If I call it editAt and move it into list.wake you still don't want to export it though right?

Reply:
You flip flopped and now I am flip flopping. You rightly refuted my $O(n^2)$ concern, since the same issues apply to plain `at`, but we trust users to use that responsibly.

Since it's the complement of `at` and you need it to implement `jAt`, ... let's just add+export it...?😑